nucypher/blockchain/eth/token.py

```diff
@@ -752,9 +752,15 @@ def _do_work(self) -> None:
         # Measure working interval
         interval = onchain_period - self.worker.last_committed_period
         if interval < 0:
-            self.__reset_tracker_state()
-            return  # No need to commit to this period.  Save the gas.
-        if interval > 0:
+            # Handle period migrations
+            last_commitment = self.worker.last_committed_period
+            next_period = onchain_period + 1
+            if last_commitment > next_period:
```

Comment:
This is to warn stakers who pointlessly commit during the 2 day migration window?

Reply:
We can't get date of upgrading automatically, this block for finding migration case (similar case led to not committing)